### ISSUE TITLE
Ability to disable ConnectionId and RequestId + RequestPath scopes

### ISSUE THREAD

**Issue by NinoFloris** [dotnet/aspnetcore#31028]
We use scopes to identify log entries as coming from a certain tenant, this is a critical part of our diagnostic ability.

Scopes have various performance downsides but due to the value they provide we really wouldn't want to disable them completely. We also add all scopes to our json logger going to cloud logging and the smaller these messages can be the better, there is a direct financial cost attached to ingestion of larger than need be messages over time.

However, and this is the issue, in the framework there exist a few scope pushes that have limited or no value for us and can't be turned off without also turning off important log messages.

`ConnectionId` is not relevant to us, but we can't turn it off because all of it is enabled under category `Microsoft.AspNetCore.Server.Kestrel` which also contains important log messages like reaching connection limits.

`RequestId` is really not relevant anymore (at least to us) with `SpanId` and `TraceId` being integrated by default but we can't turn it off
`RequestPath` is occasionaly useful but we could just as well emit it ourselves.

Our minimal scopes would ideally look like:
- Activity scope (implemented in https://github.com/dotnet/runtime/pull/37092)
- Our own scopes like tenantid, userid.

Would it be possible to provide knob(s) for these somewhere, like in `KestrelServerOptions`?

**Comment by BrennanConroy** [comment]
cc @maryamariyan 

> `RequestId` is really not relevant anymore (at least to us) with `SpanId` and `TraceId` being integrated by default but we can't turn it off

This might be redundant in the default use, but they aren't entirely useless. You can turn off Span and Trace using https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.activitytrackingoptions?view=dotnet-plat-ext-5.0

> there exist a few scope pushes that have limited or no value for us

You can provide a custom `IExternalScopeProvider` which will enable you to filter which scopes are logged.

**Comment by davidfowl** [comment]
We should support this more natively in logging. It's really not possible to turn off scopes today

**Comment by BrennanConroy** [comment]
Yep, I believe @shirhatti was going to file an issue.

**Comment by BrennanConroy** [comment]
Triage: We should look at making the `RequestId` scope configurable. And templates should have it off by default.

**Comment by NinoFloris** [comment]
@BrennanConroy this issue (and specifically that comment) led me to believe trying to override `IExternalScopeProvider` for the console logger was not a properly supported scenario https://github.com/dotnet/runtime/issues/36090#issuecomment-537177373

As to the conclusion, this means connectionid and requestpath would stay?

**Comment by markm77** [comment]
Yes, agree with this. I just tried to add OpenTelemetry context to JSON console logging in an ASP.NET Core 6 project.

Using the "IncludeScopes" formatter option I get the below scopes on a log , when all I really want is the TraceId/SpanId for correlation purposes.

```
"Scopes": [
    {
      "Message": "SpanId:7d01d402b9b9fda5, TraceId:ebd21c67aaf93cc221a6ea2ecc416a24",
      "SpanId": "7d01d402b9b9fda5",
      "TraceId": "ebd21c67aaf93cc221a6ea2ecc416a24"
    },
    {
      "Message": "ConnectionId:0HMP7SSS89CFI",
      "ConnectionId": "0HMP7SSS89CFI"
    },
    {
      "Message": "RequestPath:/x/y RequestId:0HMP7SSS89CFI:00000002",
      "RequestId": "0HMP7SSS89CFI:00000002",
      "RequestPath": "/x/y"
    },
    {
      "Message": "XXX.PostAsync (YYY)",
      "ActionId": "0e04439b-492e-4a1c-aea8-7131c3f64482",
      "ActionName": "XXX.PostAsync (YYY)"
    }

```

**Comment by tjmcdonough** [comment]
You can exclude scopes by configuring ActivityTrackingOptions when initialising the logging service.

```
builder.Services.AddLogging(options =>
{
    options.Configure(c => c.ActivityTrackingOptions = ActivityTrackingOptions.None);
});
```

**Comment by nosalan** [comment]
@tjmcdonough 
This doesn't remove the ConnectionId or RequestPath etc scopes. Only trace and span id are gone.
